engine/src/main/java/com/google/android/fhir/search/MoreSearch.kt

```diff
@@ -61,23 +61,39 @@ internal fun Search.getQuery(
   var sortOrderStatement = ""
   val sortArgs = mutableListOf<Any>()
   sort?.let { sort ->
-    val sortTableName =
+    val sortTableNames =
       when (sort) {
-        is StringClientParam -> SortTableInfo.STRING_SORT_TABLE_INFO
-        is NumberClientParam -> SortTableInfo.NUMBER_SORT_TABLE_INFO
-        is DateClientParam -> SortTableInfo.DATE_SORT_TABLE_INFO
+        is StringClientParam -> listOf(SortTableInfo.STRING_SORT_TABLE_INFO)
+        is NumberClientParam -> listOf(SortTableInfo.NUMBER_SORT_TABLE_INFO)
+        is DateClientParam ->
```

Comment:
actually add a comment here:
```suggestion
        // The DateClientParam maps to two index tables (Date without timezone info and DateTime with timezone info). Any data field in any resource will only have index records in one of the two tables. So we simply sort by both in the SQL query.
        is DateClientParam ->
```

Reply:
Yes